### PR TITLE
chore: Remove TimeSeries log scale, SummarizeTrial [WEB-1406]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

- Removes Scale parameter from CompareTrials / TimeSeries API requests, because we no longer sample linear and log requests differently
- Scale is also used in SummarizeTrial; backend asked ~1 month ago about removing this endpoint. Consensus was to remove.
- Removes Scale enum from proto
- Update tests
- For web: replace decoder type `Sdk.V1SummarizeTrialResponse` with similar `Sdk.V1ComparableTrial`

Considering this a User-facing API change which needs a release note, but it should be straightforward (don't use this param; CompareTrials replaces SummarizeTrial)

## Test Plan

- Charts continue to work correctly
- `make -C proto check` and other tests pass

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Comment by julian-determined-ai** [comment]
@mapmeld Have we confirmed that we do not want to support the `log` argument in any capacity?

**Review by julian-determined-ai — APPROVED** [review]
Front end changes look good to me!

File: docs/release-notes/7278-remove-log-option [hunk 1]
```diff
@@ -0,0 +1,7 @@
+:orphan:
```
Comment (ioga) [inline docs/release-notes/7278-remove-log-option:1]: this file should have `.rst` extension and `make fmt` applied.

**Review by ioga — APPROVED** [review]

File: docs/release-notes/7278-remove-log-option (hunk 1 above)
Comment (mapmeld) [reply docs/release-notes/7278-remove-log-option]: ✔️ thanks

**Pull request merged**